### PR TITLE
Expose IndexValidationStringency in ValidateSamFile.

### PR DESCRIPTION
**Pull request by ktibbett**

### CODE REVIEW

File: src/java/picard/sam/ValidateSamFile.java [hunk 1]
```diff
@@ -97,9 +99,17 @@
...
     public boolean IGNORE_WARNINGS = false;
 
-    @Option(doc = "If true and input is a BAM file with an index file, also validates the index.")
+    @Option(doc = "DEPRECATED.  Use INDEX_VALIDATION_STRINGENCY instead.  If true and input is " +
+            "a BAM file with an index file, also validates the index.  Until this parameter is retired " +
+            "VALIDATE INDEX and INDEX_VALIDATION_STRINGENCY must agree on whether to validate the index. ")
     public boolean VALIDATE_INDEX = true;
 
+    @Option(doc = "If set to anything other than IndexValidationStringency.NONE and input is " +
+            "a BAM file with an index file, also validates the index at the specified stringency." +
```
Comment (nh13) [inline src/java/picard/sam/ValidateSamFile.java]: missing space after `.`.

File: src/java/picard/sam/ValidateSamFile.java [hunk 2]
```diff
@@ -97,9 +99,17 @@
     @Option(doc = "If true, only report errors and ignore warnings.")
     public boolean IGNORE_WARNINGS = false;
 
-    @Option(doc = "If true and input is a BAM file with an index file, also validates the index.")
+    @Option(doc = "DEPRECATED.  Use INDEX_VALIDATION_STRINGENCY instead.  If true and input is " +
+            "a BAM file with an index file, also validates the index.  Until this parameter is retired " +
+            "VALIDATE INDEX and INDEX_VALIDATION_STRINGENCY must agree on whether to validate the index. ")
```
Comment (nh13) [inline src/java/picard/sam/ValidateSamFile.java]: Extra space after `.`.

File: src/java/picard/sam/ValidateSamFile.java [hunk 3]
```diff
@@ -185,4 +195,18 @@ protected int doWork() {
 
         return result ? 0 : 1;
     }
+
+    @Override
+    protected String[] customCommandLineValidation() {
+        if ((!VALIDATE_INDEX  && INDEX_VALIDATION_STRINGENCY != IndexValidationStringency.NONE) ||
```
Comment (nh13) [inline src/java/picard/sam/ValidateSamFile.java]: two spaces before `&&`

File: src/java/picard/sam/ValidateSamFile.java [hunk 4]
```diff
@@ -185,4 +195,18 @@ protected int doWork() {
...
+            (INDEX_VALIDATION_STRINGENCY == IndexValidationStringency.NONE && VALIDATE_INDEX)) {
```
Comment (nh13) [inline src/java/picard/sam/ValidateSamFile.java]: to be consistent with the previous line, could you put `VALIDATE_INDEX` before `INDEX_VALIDATION_STRINGENCY == IndexValidationStringency.NONE`?

File: src/java/picard/sam/ValidateSamFile.java [hunk 5]
```diff
@@ -185,4 +195,18 @@ protected int doWork() {
...
+            return new String[]{"VALIDATE_INDEX and INDEX_VALIDATION_STRINGENCY must be consistent: " +
+                    "VALIDATE_INDEX is " + VALIDATE_INDEX + " and INDEX_VALIDATION_STRINGENCY is " +
+                    INDEX_VALIDATION_STRINGENCY};
+        }
+
+        return super.customCommandLineValidation();
+    }
+
+
```
Comment (nh13) [inline src/java/picard/sam/ValidateSamFile.java]: extra newlines.

File: src/java/picard/sam/ValidateSamFile.java [hunk 6]
```diff
@@ -29,9 +29,11 @@
 import htsjdk.samtools.SamReader;
 import htsjdk.samtools.SamReaderFactory;
 import htsjdk.samtools.ValidationStringency;
+import htsjdk.samtools.BamIndexValidator.IndexValidationStringency;
 import htsjdk.samtools.reference.ReferenceSequenceFile;
 import htsjdk.samtools.reference.ReferenceSequenceFileFactory;
 import htsjdk.samtools.util.IOUtil;
+import htsjdk.tribble.index.Index;
```
Comment (nh13) [inline src/java/picard/sam/ValidateSamFile.java]: what's this import for?

**Comment by ktibbett** [comment]
@nh13 , comments addressed.  @vivster7 , could you please take a look at this as well?

**Comment by vivster7** [comment]
squash commits, then :+1:

File: src/java/picard/sam/ValidateSamFile.java (hunk 6 above)
Comment (ktibbett) [inline src/java/picard/sam/ValidateSamFile.java]: sorry, missed this one.

**Pull request merged**